src/solvers/lowering/byte_operators.cpp

```diff
@@ -475,26 +475,17 @@ static exprt unpack_array_vector(
         body};
     }
 
-    optionalt<exprt> array_vector_size;
-    optionalt<typet> subtype;
-    if(src.type().id() == ID_vector)
-    {
-      array_vector_size = to_vector_type(src.type()).size();
-      subtype = to_vector_type(src.type()).subtype();
-    }
-    else
-    {
-      array_vector_size = to_array_type(src.type()).size();
-      subtype = to_array_type(src.type()).subtype();
-    }
+    const exprt array_vector_size = src.type().id() == ID_vector
+                                      ? to_vector_type(src.type()).size()
+                                      : to_array_type(src.type()).size();
```

Comment:
Good refactor, but would prefer if this was a separate commit

Reply:
Done!